src/data/list/nodup.lean

```diff
@@ -127,6 +127,23 @@ theorem nodup_map_on {f : α → β} {l : list α} (H : ∀x∈l, ∀y∈l, f x
   (d : nodup l) : nodup (map f l) :=
 pairwise_map_of_pairwise _ (by exact λ a b ⟨ma, mb, n⟩ e, n (H a ma b mb e)) (pairwise.and_mem.1 d)
 
+theorem inj_on_of_nodup_map {f : α → β} {l : list α} (d : nodup (map f l)) :
+  ∀ (x ∈ l) (y ∈ l), f x = f y → x = y :=
```

Comment:
```suggestion
  set.inj_on f s :=
```

Reply:
The remarks about binder implicitness apply to this lemma too, although here `set.inj_on` isn't possible.

Reply:
In case you missed this, @bhavik